### PR TITLE
Add resolver for accountInfo

### PR DESCRIPTION
**Pull request by henrytso**
Makes three requests upon resolving.
Created TransactionType and AccountInfoType.
Final response will be of AccountInfoType.

The 'id' of the query is a user's sessionId.
A sessionId can be obtained from:
[https://get.cbord.com/cornell/full/login.php?mobileapp=1](url)

Sample query:
```
{
  accountInfo(id: "a625b478-dbe9-4810-968b-06eeb34e14b4") {
    cityBucks
    laundry
    brbs
    swipes
    history {
      name
      timestamp
    }
  }
}
```

### CODE REVIEW

File: schema.py [hunk 1]
```diff
@@ -139,3 +163,84 @@ def resolve_eateries(self, info, eatery_id=None):
...
+            }
+        }
+    ).json()['response']['accounts']
+
+    for acct in accounts:
+      if acct['accountDisplayName'] == ACCOUNT_NAMES['citybucks']:
+        account_info['cityBucks'] = str(acct['balance'])
+      if acct['accountDisplayName'] == ACCOUNT_NAMES['laundry']:
+        account_info['laundry'] = str("{0:.2f}".format(round(acct['balance'],2)))
+      if acct['accountDisplayName'] == ACCOUNT_NAMES['brbs']:
```
Comment (42B) [inline schema.py]: Couldn't these last two be `elif`'s ?

File: schema.py [hunk 2]
```diff
@@ -139,3 +163,84 @@ def resolve_eateries(self, info, eatery_id=None):
...
+    # Query 3: Get list of transactions
+    CORNELL_INSTITUTION_ID = '73116ae4-22ad-4c71-8ffd-11ba015407b1'
+    transactions = requests.post(commerce_url,
+        json={
+            'method': 'retrieveTransactionHistory',
+            'params': {
+                'paymentSystemType': 0,
+                'queryCriteria': {
+                    'accountId': None,
+                    'endDate': datetime.now().isoformat()[:10],
```
Comment (42B) [inline schema.py]: Another way?

```python
In[2]: from datetime import datetime
In[3]: datetime.now().isoformat()
Out[3]: '2018-11-03T22:20:02.452388'
In[4]: datetime.now().isoformat()[:10]
Out[4]: '2018-11-03'

In[5]: datetime.now().date()
Out[5]: datetime.date(2018, 11, 3)
In[6]: str(datetime.now().date())
Out[6]: '2018-11-03'
```

File: schema.py [hunk 3]
```diff
@@ -139,3 +163,84 @@ def resolve_eateries(self, info, eatery_id=None):
...
+            'version': '1'
+        }
+    ).json()['response']['transactions']
+
+    account_info['history'] = []
+
+    for t in transactions:
+      date = t['actualDate'][:10]
+      time = t['actualDate'][11:16]
+      time = datetime.strptime(time, "%H:%M").strftime("%I:%M %p")
```
Comment (ilikesymmetry) [inline schema.py]: have a comment that explains this line

File: schema.py [hunk 4]
```diff
@@ -1,6 +1,15 @@
...
 from graphene.types.datetime import Date, Time
+import requests
+from datetime import datetime
 
+ACCOUNT_NAMES = {
+    'brbs': 'BRB Big Red Bucks',
+    'citybucks': 'CB1 City Bucks',
+    'laundry':  'LAU Sem Laundry',
+    'cornell_card': 'CC1 Cornell Card'
+    # 'swipes': ''
```
Comment (young-k) [inline schema.py]: delete this?

File: schema.py [hunk 5]
```diff
@@ -1,6 +1,15 @@
 from graphene import Field, ObjectType, String, List, Int, Float, Boolean
 from graphene.types.datetime import Date, Time
+import requests
+from datetime import datetime
```
Comment (young-k) [inline schema.py]: alphabetize this

File: schema.py [hunk 6]
```diff
@@ -1,6 +1,15 @@
...
 
+ACCOUNT_NAMES = {
+    'brbs': 'BRB Big Red Bucks',
+    'citybucks': 'CB1 City Bucks',
+    'laundry':  'LAU Sem Laundry',
+    'cornell_card': 'CC1 Cornell Card'
```
Comment (young-k) [inline schema.py]: alphabetize as well

File: schema.py [hunk 7]
```diff
@@ -116,6 +125,17 @@ def __init__(self, **kwargs):
...
 
+class TransactionType(ObjectType):
+  name = String(required=True)
+  timestamp = String(required=True)
+
+class AccountInfoType(ObjectType):
+  cityBucks = String(required=True)
+  laundry = String(required=True)
+  brbs = String(required=True)
+  swipes = String(required=True)
```
Comment (young-k) [inline schema.py:97]: alphabetize fields

File: schema.py [hunk 8]
```diff
@@ -139,3 +163,84 @@ def resolve_eateries(self, info, eatery_id=None):
...
+
+  def resolve_account_info(self, info, session_id=None):
+    if session_id is None:
+      return "Provide a valid session ID!"
+
+    account_info = {}
+
+    # URLs to send requests to
+    user_url = 'https://services.get.cbord.com/GETServices/services/json/user'
+    commerce_url = 'https://services.get.cbord.com/GETServices/services/json/commerce'
```
Comment (ilikesymmetry) [inline schema.py]: put these urls in constants.py

File: schema.py [hunk 9]
```diff
@@ -130,6 +150,10 @@ class Query(ObjectType):
       eatery_id=Int(name='id'),
       date=Date()
   )
+  account_info = Field(AccountInfoType,
+      session_id=String(name='id'),
+      date=Date()
```
Comment (young-k) [inline schema.py]: alphabetize here as well

File: schema.py [hunk 10]
```diff
@@ -139,3 +163,84 @@ def resolve_eateries(self, info, eatery_id=None):
...
+      if acct['accountDisplayName'] == ACCOUNT_NAMES['laundry']:
+        account_info['laundry'] = str("{0:.2f}".format(round(acct['balance'],2)))
+      if acct['accountDisplayName'] == ACCOUNT_NAMES['brbs']:
+        account_info['brbs'] = str(acct['balance'])
+      # Need more research to implement swipes:
+      # Each plan has a different accountDisplayName
+      account_info['swipes'] = ''
+
+    # Query 3: Get list of transactions
+    CORNELL_INSTITUTION_ID = '73116ae4-22ad-4c71-8ffd-11ba015407b1'
```
Comment (ilikesymmetry) [inline schema.py:164]: put in constants.py

File: schema.py (hunk 8 above)
Comment (young-k) [inline schema.py]: can we put the common prefix in a constant at the top?

File: schema.py (hunk 1 above)
Comment (young-k) [reply schema.py]: +1

File: schema.py (hunk 2 above)
Comment (young-k) [reply schema.py]: +1 to this

**Review by young-k — CHANGES_REQUESTED** [review]
LGTM, minor changes

File: schema.py (hunk 10 above)
Comment (young-k) [reply schema.py:164]: @henrytso this should be removed from here?

**Review by young-k — APPROVED** [review]
LGTM, address the last comment. In the future, your branch should be named `henry/brbs`, check out bible.cornellappdev.com for our Git guide!

**Pull request merged**